main.go

```diff
@@ -83,6 +84,9 @@ func main() {
 	// init log format from envvar ESTAFETTE_LOG_FORMAT
 	foundation.InitLoggingFromEnv(foundation.NewApplicationInfo(appgroup, app, version, branch, revision, buildDate))
 
+	// create context to cancel commands on sigterm
+	ctx := foundation.InitCancellationContext(context.Background())
```

Comment:
Why is this required for the cloudflare dns? This is suppose to be running as a controller in the background. Do we need to cancel the contexts on terminations? sigterm triggers the cancel function and for a controller which is a long running object, I think its not a good practice to include this

Reply:
I think we can do it either way with or without canceling context, but this is case it cannot reach the kube API for some reason. It gives a safeguard timeouts. What do you think? Keep in mind that this is only for the kube API comms, not CloudFlare DNS.

Reply:
We are already doing graceful shutdown of the application level : https://github.com/estafette/estafette-cloudflare-dns/blob/main/main.go#L180. So for the kube-api we are hadling this in a way

Reply:
@ivanovaleksandar I think its fine to add this. Please test the changes in playground to catch any unexpected behaviours